otx/algorithms/classification/adapters/mmcls/nncf/builder.py

```diff
@@ -0,0 +1,170 @@
+    else:
+        # pytorch ckpt
+        state_to_build_nncf = state_dict
+
+        # This data and state dict will be used to build NNCF graph later
+        # when loading NNCF model
+        # because some models run their subcomponents based on intermediate outputs
+        # resulting differently and partially traced NNCF graph
+        datasets = get_configs_by_keys(config.data.train, "otx_dataset")
+        data_to_build_nncf = datasets[0][0].numpy
+
+        init_dataloader = build_dataloader(
+            config,
+            subset="train",
+            distributed=distributed,
+            dataloader_builder=mmcls_build_dataloader,
+            dataset_builder=build_dataset,
+        )
+
+        val_dataloader = None
+        if is_acc_aware:
+            val_dataloader = build_dataloader(
+                config,
+                subset="val",
+                distributed=distributed,
+                dataloader_builder=mmcls_build_dataloader,
+                dataset_builder=build_dataset,
+            )
+
+        model_eval_fn = partial(
+            model_eval,
+            config=config,
+            val_dataloader=val_dataloader,
+            evaluate_fn=multi_gpu_test if distributed else single_gpu_test,
+            distributed=distributed,
+        )
+        state_dict = None
+    load_state_dict(model, state_to_build_nncf)
+
+    test_pipeline = Compose(config.data.test.pipeline)
+    get_fake_input_fn = partial(
+        get_fake_input,
+        preprocessor=test_pipeline,
+        data=data_to_build_nncf,
+    )
+
+    compression_ctrl, model = wrap_nncf_model(
+        config,
+        model,
+        init_state_dict=state_dict,
+        model_eval_fn=model_eval_fn,
+        get_fake_input_fn=get_fake_input_fn,
+        dataloader_for_init=init_dataloader,
+        is_accuracy_aware=is_acc_aware,
+    )
+
+    # update custom hooks
+    custom_hooks = config.get("custom_hooks", [])
+    custom_hooks.append(
+        ConfigDict(type="CompressionHook", compression_ctrl=compression_ctrl)
+    )
+    custom_hooks.append(ConfigDict({"type": "CancelTrainingHook"}))
+    custom_hooks.append(
+        ConfigDict(
+            type="CheckpointHookBeforeTraining",
+            save_optimizer=True,
+            meta={
+                DATA_TO_BUILD_NAME: data_to_build_nncf,
+                STATE_TO_BUILD_NAME: state_to_build_nncf,
+            },
+        )
+    )
+    remove_from_configs_by_type(custom_hooks, "CancelInterfaceHook")
+    remove_from_configs_by_type(custom_hooks, "TaskAdaptHook")
+    remove_from_configs_by_type(custom_hooks, "AdaptiveTrainSchedulingHook")
+
+    for hook in get_configs_by_dict(custom_hooks, dict(type="OTXProgressHook")):
+        time_monitor = hook.get("time_monitor", None)
+        if (
+            time_monitor
+            and getattr(time_monitor, "on_initialization_end", None) is not None
+        ):
+            time_monitor.on_initialization_end()
```

Comment:
What is the purpose of this block?

Reply:
This block is for [this callback](https://github.com/openvinotoolkit/training_extensions/blob/feature/otx-new-mmcv/base/otx/algorithms/common/tasks/nncf_base.py#L277-L281). `OptimizationProgressCallback` has an additional method [`on_initialization_end`](https://github.com/openvinotoolkit/training_extensions/blob/9002dda17ae0801fadf35990ed2bf86500a78b31/otx/algorithms/common/utils/callback.py#L117-L121), which needs to be called after initialization of NNCF model.